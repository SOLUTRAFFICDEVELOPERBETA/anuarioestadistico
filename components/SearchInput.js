import React from 'react';
import PropTypes from 'prop-types';
import { TextField, InputAdornment } from '@material-ui/core';
import { Search } from '@material-ui/icons';
import { useForm } from 'react-hook-form';

/**
 * @description Componente que permite la busqueda de un elemento de una lista.
 * @param {func: ()=> void} onSearch
 * @param {String} placeholder 
 * Consulte los elementos utilizados en {@link https://material-ui.com/ | Material-ui}
 */
const SearchInput = ({ onSearch, placeholder }) => {
    const { register, handleSubmit } = useForm();

    return (
        <form onSubmit={handleSubmit(onSearch)}>
            <TextField
                fullWidth
                autoComplete="search-input"
                name="search"
                placeholder={placeholder}
                margin="dense"
                variant="outlined"
                inputRef={register}
                InputProps={{
                    endAdornment: (
                        <InputAdornment position="end">
                            <Search color="primary" />
                        </InputAdornment>
                    )
                }}
            />
        </form>
    );
};

SearchInput.propTypes = {
    onSearch: PropTypes.func,
    placeholder: PropTypes.string
};

SearchInput.defaultProps = {
    onSearch: (search) => console.log(search),
    placeholder: 'Buscar...'
};

export default SearchInput;
