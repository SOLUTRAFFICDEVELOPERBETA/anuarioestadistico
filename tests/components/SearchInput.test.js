import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import SearchInput from '../../components/SearchInput';

describe('Pruebas en el componente <SearchInput/>', () => {
    const onSearch = jest.fn()
    const wrapper = shallow(
        <SearchInput
            onSearch={onSearch}
            label={'label de prueba'}
        />
    )

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe llamar la funciÃ³n, onSearch()', () => {
        wrapper.find('WithStyles(ForwardRef(TextField))').prop('inputRef')()
        wrapper.find('form').simulate('submit');

        expect(onSearch).toHaveBeenCalled();
        expect(onSearch).toHaveBeenCalledTimes(1)
    })

})