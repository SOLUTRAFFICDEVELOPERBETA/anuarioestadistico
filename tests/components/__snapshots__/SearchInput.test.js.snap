// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <SearchInput/> debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <WithStyles(ForwardRef(TextField))
    InputProps={
      Object {
        "endAdornment": <WithStyles(ForwardRef(InputAdornment))
          position="end"
        >
          <Memo(SearchIcon)
            color="primary"
          />
        </WithStyles(ForwardRef(InputAdornment))>,
      }
    }
    autoComplete="search-input"
    fullWidth={true}
    inputRef={[Function]}
    margin="dense"
    name="search"
    placeholder="Buscar..."
    variant="outlined"
  />
</form>
`;
